core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -160,9 +165,8 @@ public void run() {
           resolving = true;
         }
         try {
-          InetSocketAddress destination = InetSocketAddress.createUnresolved(host, port);
-          ProxyParameters proxy = proxyDetector.proxyFor(destination);
```

Comment:
We want the name resolver doing the proxy detection. As part of #4029 we'll start allowing the name resolver to return proxy information.

Reply:
#4029 is closed, but it seems we don't yet provide the ProxyDetector to the NameResolver. I'm thinking that can be passed in in the `Attributes` passed to `newNameResolver()`.